### DISCUSSION TITLE
Use variable in yaml

### DISCUSSION THREAD

**Discussion by zdary** [codemagic-ci-cd/codemagic-docs#2631] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

I'd like to use current git branch name which triggered the build in codemagic.yaml file

workflows:
  react-native-android:
    name: AU "${CM_BRANCH}" Android

The variable here doesn't resolve. 

Another place I need the current branch is in script. We have .env.<BRANCHNAME> files in our project. We need to read variables from branch specific file.
I tried to follow the example from here https://docs.codemagic.io/knowledge-others/import-variables-from-env-file/

    scripts:
      - name: Set value from "${CM_BRANCH}" "$CM_BRANCH"
        script: |
          
          source ${.env.CM_BRANCH}

<img width="1111" alt="2024-07-11_21-55-05" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/3831870/c0d32233-a766-40ca-9481-d60584e3776d">

### Expected results

name should resolve to: AU dev Android
source ${.env.CM_BRANCH} should resolve to source .env.dev

### Actual results

branch name is not resolved into the build name: AU "${CM_BRANCH}" Android
source ${.env.CM_BRANCH} is not resolved

### Build id (optional)

668fc7f66b9addc2a5a90438

**Comment by himesh-cm** [comment, marked as answer]
Hey @zdary,

Please find here with below sample script for sourcing env vars from `.env.dev` file.

```
      - name: fetch env vars from
        script: |
          BRANCH_ENV_FILE=".env.${CM_BRANCH}"
          echo $BRANCH_ENV_FILE
          if [ -f "$BRANCH_ENV_FILE" ]; then
            source "$BRANCH_ENV_FILE"
            echo "FOO=$FOO" >> $CM_ENV
            echo $FOO
          else
            echo "Environment file for branch ${CM_BRANCH} not found."
          fi
```
Currently, the script name and workflow name resolve to a string value rather than dynamically fetching it from an env var and setting it when you run the workflow. This is expected.

**Reply by zdary** [reply]
thank you @himesh-cm much appreciated. 
Any plans to support the name resolve in the future? This would prevent a lot of code duplication.

**Reply by himesh-cm** [reply]
Hey @zdary, please feel free to raise your request under "Ideas" at https://github.com/orgs/codemagic-ci-cd/discussions/categories/ideas to see if there is interest from others as well.

**Reply by himesh-cm** [reply]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
